src/file_ext/async_impl.rs

```diff
@@ -42,20 +42,54 @@ macro_rules! async_file_ext {
             /// locked exclusively.
             fn lock_shared(&self) -> Result<()>;
 
+            /// Locks the file for exclusive usage, blocking if the file is currently
+            /// locked exclusively.
+            ///
+            /// **Note:** This method is not really "async", the underlying system call is still blocking.
+            /// Having this method as async is just for convenience when using it in async runtime.
```

Comment:
This is NOT convenient at all. This is in fact very misleading and confusing to call an async function and get runtime blocked!

Consider remove all APIs that are not actually async to prevent such confusion. This will allow users to use `Arc<File>` in something like `tokio::task::spawn_blocking()` explicitly preventing runtime blocking.